packages/code-du-travail-data/indexing/cdtnIds.js

```diff
@@ -0,0 +1,49 @@
+import * as XXH from "xxhashjs";
+
+// never change this seed, as ids are regenerated
+// each time but must remain stable
+const H = XXH.h64(0x1e7f);
+
+// default id length, can be longer (up to 16)
+const maxIdLength = 10;
+
+// use xxhash to hash source+id
+const hashId = (source, id, idLength) =>
+  // save 64bits hash as Hexa string up to maxIdLength chars (can be changed later in case of collision)
+  // as the xxhash function ensure distribution property
+  H.update(source + id)
+    .digest()
+    .toString(16)
+    .slice(0, idLength);
+
+const hashingSet = (hashMap, idLength) => (obj) => {
```

Comment:
Alors ici, si tu veux juste une fonction avec des params par défault, plutôt que de faire une fonction qui appelle un fonction avec les paramètres (createHashingSet), c'est plus commun de juste faire ça: 

```suggestion
const hashingSet = (hashMap = new Map(), idLength = maxIdLength) => (obj) => {
```

Reply:
Ensuite, je ne pense pas que dans l'immediat, tu aies besoin de mettre `hashMap` en param parce que je ne vois pas trop pourquoi quelqu'un aurait besoin de fournir sa propre map dans l'immediat ?

Reply:
en fait la hash map c'est pour conserver la liste des ids déjà attribué, et ainsi [détecter une collision](https://github.com/SocialGouv/code-du-travail-numerique/pull/2878/files#diff-420468db4171239d04509af6abce13bdR27)

Reply:
Yes je ne doute pas de son utilité, je pensais plus à quelque chose comme ça:
Mais alors c'est vraiment juste pour chipoter hein
```

(idLength = maxIdLength) => {
  const hashMap = new Map();
  return (obj) => { la foonction };
}
```

Reply:
ah oui j'ai lu ton comment trop vite, en fait c'est dans le test que je passe la map en param

Reply:
oook then ! 👍